Detectors/TPC/simulation/src/ClustererTask.cxx

```diff
@@ -70,32 +68,35 @@ InitStatus ClustererTask::Init()
     return kERROR;
   }
 
-  mDigitsArray = dynamic_cast<TClonesArray *>(mgr->GetObject("TPCDigit"));
+  mDigitsArray = mgr->InitObjectAs<decltype(mDigitsArray)>("TPCDigit");
   if( !mDigitsArray ) {
     LOG(ERROR) << "TPC points not registered in the FairRootManager. Exiting ..." << FairLogger::endl;
     return kERROR;
   }
 
   if (mBoxClustererEnable) {
-    mBoxClusterer = new BoxClusterer();
-    mBoxClusterer->Init();
     
-    // Register output container
-    mClustersArray = new TClonesArray("o2::TPC::Cluster");
-    mgr->Register("TPCCluster", "TPC", mClustersArray, kTRUE);
+    // Create and register output container
+    mClustersArray = new std::vector<o2::TPC::BoxCluster>;
```

Comment:
Does this have to be on the heap after this change or can it also be on the stack? Does the Manager own this object?
Same below for the box clusterer.

Reply:
Same answer as previously: At the moment neither stack based nor smart pointers work together with the FairRootManager, which is why it has to stay as it is for the moment.

I agree that this should be improved.